Interface Class Specifications/Display/InterfaceSpec-Display.md

```diff
@@ -0,0 +1,86 @@
+
+**From other classes**: Although not expected to be common, it may be possible to derive a display interface out of a set of other interfaces representing individual parameters.
+
+**To other classes**: If some display parameters are dynamic, and the hardware in question can read the value of the parameter at runtime as it changes, then it may be useful to expose that parameter separately as its own interface. For example, a display's center of projection might be exposed as an `OSVR_Vec2` interface.
+
+## Messages
+In general, to work smoothly with the rest of the OSVR system, distance units are meters and velocity units are meters/second.
+
+### Display
+#### Data
+- type (enum in fixed|hmd)
+- vendor (string)
+- model: (string)
+- product URI (uri-string RFC 3986)
+- driver URI (uri-string RFC 3986)
+- notes: string
+- number of DisplayInputs (whole number)
+- number of logical DisplaySurfaces (whole number)
+- k1[3] rgb radial distortion supported (boolean, hmd only?) and priority (whole number)
+- RGB radial distortion supported (boolean, hmd only?, supercedes k1[3] in most cases) and priority (whole number)
+- mono point mesh distortion supported (boolean, hmd only?) and priority (whole number)
+- RGB mesh distortion supported (boolean, hmd only?) and priority (whole number)
+
+#### Rationale
+The Display message is the root object of the Display interface. It references both the physical DisplayInputs and logical DisplaySurfaces, which define the mapping between them.
+
+### DisplayInput
+> one per number of display inputs in Display message.
+
+#### Data
+- resolution (width, height in whole numbers)
+- vendor PNPID (string)
+- product id (string)
+- platform-specific hardware instance path to uniquely identify it (??? string?)
+- if part of the desktop, origin on the desktop (X and Y in whole numbers)
+- On Windows, dx adapter number (??? string?)
+- logical id or platform-agnostic identifier (string? or int? referenced by logical topology)
+- location and orientation (physical orientation and location? relative to what?)
+- eye relief/lense-to-screen distance (if not represented with location/orientation above)
+- Scan-out direction (unit OSVR_Vec2)
+- Scan-out framerate (analog/float)
+- estimated latency to start of render in ms (analog/float)
+
+#### Rationale
+The DisplayInput message represents a physical display input.
+
+### DisplaySurface
+> one per number of logical display surfaces in Display message
+
+#### Data
+- logical id of display input for this surface (from DisplayInput message)
+- eye index (whole number)
+- viewer index (whole number)
+- viewport (relative to this surface's corresponding DisplayInput)
+- rotation (enum in landscape|portrait|landscape-flipped|portrait-flipped in clockwise order, relative to corresponding DisplayInput)
```

Comment:
So the rotation value here replaces the current swap_180 flag?

I'd like to make a distinction between rotation and orientation. Rotations in my mind are relative and 'stackable'—that is, I can continue adding rotations and I'll end up with a final orientation. But if a DisplayInput has a landscape orientation and a DisplaceSurface has a portrait-flipped orientation, I don't know what that means.

This may just be a problem with my understanding of these things. But when Windows reports an orientation in its settings that orientation doesn't necessarily correspond to the expected physical appearance on the display and this was driving me batty while working on the display rotation stuff for the SteamVR driver.